From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Thu, 11 Oct 2018 08:49:00 +0200
Subject: [PATCH] dsa: ksz: dont complain about not being able to remove entry

The entries are added without being verified if added multiple times
but get removed multiple times. On the second removal the entry will
not be found and the driver complains.

Signed-off-by: Michael Grzeschik <m.grzeschik@pengutronix.de>
---
 drivers/net/dsa/microchip/ksz_common.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/dsa/microchip/ksz_common.c b/drivers/net/dsa/microchip/ksz_common.c
index ff3b614d3d86..d3d4adffe130 100644
--- a/drivers/net/dsa/microchip/ksz_common.c
+++ b/drivers/net/dsa/microchip/ksz_common.c
@@ -349,10 +349,8 @@ int ksz_port_mdb_del(struct dsa_switch *ds, int port,
 	}
 
 	/* no available entry */
-	if (index == dev->num_statics) {
-		ret = -EINVAL;
+	if (index == dev->num_statics)
 		goto exit;
-	}
 
 	/* clear port */
 	alu.port_forward &= ~BIT(port);
